### PR TITLE
Install operator at a fixed version

### PR DESCRIPTION
**Pull request by SaphMB**
## Expected behaviour for acceptance

**Default with no attempt to set the `tag`**

Context: A `Chart.yaml` where appVersion = v0.3.1 (reflecting the Operator version). `imageRegistry.tag` is not set.

_Setup cluster with cert-manager_
```
// from helm-charts/ske-operator

kind create cluster

kubectl apply --filename https://github.com/cert-manager/cert-manager/releases/download/v1.15.0/cert-manager.yaml
namespace/cert-manager
```

_Update `values.yaml`_

```
skeLicense: "a-real-token"
```

_Install Chart with `values.yaml`_
```
helm install ske-operator . \
  --namespace kratix-platform-system \
  --create-namespace \
  --values values.yaml
```

_Check operator version_
```
helm list --all-namespaces

-> ske-operator	kratix-platform-system	1       	2024-10-16 12:31:55.732419 +0100 BST	deployed	ske-operator-0.17.0	v0.3.1
```

SKE Operator version should match the `appVersion` in `Chart.yaml`

```
k get pods --selector=control-plane=ske-operator -A -o yaml | yq '.items[0].spec.containers[0].image'

-> ghcr.io/syntasso/ske-operator:v0.3.1
```

Image tag should match the `appVersion` in `Chart.yaml`

**Attempting to set the deprecated `tag`**

Context: A `Chart.yaml` where appVersion = v0.3.1 (reflecting the Operator version). `imageRegistry.tag` is set to `v0.2.0`.

tl;dr same behaviour with the tag ignored

_Setup cluster with cert-manager_
```
// from helm-charts/ske-operator

kind create cluster

kubectl apply --filename https://github.com/cert-manager/cert-manager/releases/download/v1.15.0/cert-manager.yaml
namespace/cert-manager
```

_Update `values.yaml`_

```
skeLicense: "a-real-token"
```

_Install Chart with `values.yaml`_
```
helm install ske-operator . \
  --namespace kratix-platform-system \
  --create-namespace \
  --values values.yaml
```

_Check operator version_
```
helm list --all-namespaces

-> ske-operator	kratix-platform-system	1       	2024-10-16 12:31:55.732419 +0100 BST	deployed	ske-operator-0.17.0	v0.3.1
```

SKE Operator version should match the `appVersion` in `Chart.yaml`

```
k get pods --selector=control-plane=ske-operator -A -o yaml | yq '.items[0].spec.containers[0].image'

-> ghcr.io/syntasso/ske-operator:v0.3.1
```

Image tag should match the `appVersion` in `Chart.yaml`

### CODE REVIEW

File: ske-operator/templates/operator-config.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ data:
         {{- with .Values.releaseStorage.bucket }}
         name: {{ .name }}
         endpoint: {{ .endpoint }}
-        region: {{ .rergion }}
+        region: {{ .region }}
```
Comment (aclevername) [inline ske-operator/templates/operator-config.yaml:26]: 😂

**Review by aclevername — APPROVED** [review]
